source/common/protobuf/utility.h

```diff
@@ -42,6 +42,17 @@
        ? Protobuf::util::TimeUtil::DurationToSeconds((message).field_name())                       \
        : throw MissingFieldException(#field_name, (message)))
 
+// Convert an envoy::api::v2::core::Percent to a rounded integer or a default.
+// @param message supplies the proto message containing the field.
+// @param field_name supplies the field name in the message.
+// @param max_value supplies the maximum allowed integral value (e.g., 100, 10000, etc.).
+// @param default_value supplies the default if the field is not present.
+#define PROTOBUF_PERCENT_TO_ROUNDED_INTEGER_OR_DEFAULT(message, field_name, max_value,             \
```

Comment:
The macro resolves to a double value, e.g. not rounded and not an integer.

Reply:
Good point I will add an explicit cast.

Reply:
Fixed.

Reply:
If we want this to be a general purpose utility should we add guards/ASSERTs to avoid overflow?

I'm nowhere as worried about config causing unexpected behavior as malicious users causing unexpected behavior, so I'm OK if the answer is that we should leave this for now and assume it'll be handled as part of the general audit when @htuch does config fuzzing.

Reply:
Sure I can add some ASSERTs.